anvil/src/config.rs

```diff
@@ -531,15 +531,15 @@ impl NodeConfig {
         let mut env = revm::Env {
             cfg: CfgEnv {
                 spec_id: self.hardfork.into(),
-                chain_id: self.chain_id.into(),
+                chain_id: self.chain_id.unwrap_or(CHAIN_ID).into(),
                 ..Default::default()
             },
             block: BlockEnv {
                 gas_limit: self.gas_limit,
                 basefee: self.get_base_fee(),
                 ..Default::default()
             },
-            tx: TxEnv { chain_id: Some(self.chain_id), ..Default::default() },
+            tx: TxEnv { chain_id: self.chain_id, ..Default::default() },
```

Comment:
this could end up being `None`

so need to also use the new chain_id getter here